packages/toolpad-app/src/toolpadDataSources/sql/server.ts

```diff
@@ -0,0 +1,36 @@
+import { errorFrom } from '../../utils/errors';
+import { Maybe } from '../../utils/types';
+import { SqlConnectionParams, SqlExecBase, SqlPrivateQuery, SqlQuery, SqlResult } from './types';
+
```

Comment:
I was wondering whether it would make sense to abstract the whole datasource?
```tsx
export function createSqlServerDatasource<...>({ execSql, tablesQuery }): ServerDatasource<...> {
  // ...
}
```

Reply:
Agreed. I've pushed a change that does this